tensorbay/opendataset/CACD/loader.py

```diff
@@ -0,0 +1,126 @@
+
+import os
+from typing import Any, Dict, Tuple
+
+import numpy as np
+
+from ...dataset import Data, Dataset
+from ...exception import ModuleImportError
+from ...label import CategoryInfo, Classification, ClassificationSubcatalog
+from ...utility import NameList
+
+DATASET_NAME = "CACD"
+_ATTRIBUTES = {
+    "attributes": [
+        {"type": "number", "minimum": 14, "maximum": 62, "name": "age"},
+        {"type": "number", "minimum": 1951, "maximum": 1990, "name": "birth"},
+        {"type": "number", "minimum": 1, "maximum": 50, "name": "range"},
+        {"type": "number", "minimum": 2004, "maximum": 2013, "name": "year"},
+        {"type": "string", "name": "name"},
+        {
+            "type": "boolean",
+            "name": "lfw",
+            "description": "Whether the celebrity is in LFW dataset.",
+        },
+    ]
+}
+
+
+def CACD(path: str) -> Dataset:
+    """Dataloader of `Cross-Age Celebrity Dataset (CACD)`_ dataset.
+
+    .. _Cross-Age Celebrity Dataset (CACD): https://bcsiriuschen.github.io/CARC/
+
+    The file structure should be like::
+
+        <path>
+            CACD2000/
+                14_Aaron_Johnson_0001.jpg
+                ...
+            celebrity2000.mat
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+    """
+    dataset = Dataset(DATASET_NAME)
+    dataset.catalog.classification = _get_catalog()
+    segment = dataset.create_segment()
+    img_files = sorted(os.listdir(os.path.join(path, "CACD2000")))
+    labels_map = _get_labels_map(os.path.join(path, "celebrity2000.mat"))
+    for img in img_files:
+        category, attribute = labels_map[img]
+        img_data = Data(os.path.join(path, "CACD2000", img))
+        img_data.label.classification = Classification(category, attribute)
+        segment.append(img_data)
+    return dataset
+
+
+def _get_labels_map(path: str) -> Dict[str, Tuple[str, Dict[str, Any]]]:
+    """Get celebrity_data from .mat file.
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Raises:
+        ModuleImportError: When the module "h5py" can not be found.
+
+    Returns:
+        A Dict of attributes.
+
+    """
+    try:
+        from h5py import File  # pylint: disable=import-outside-toplevel
+    except ModuleNotFoundError as error:
+        raise ModuleImportError(error.name) from error  # type: ignore[arg-type]
+    mat_file = File(path)
+    celebrity_data = mat_file["celebrityImageData"]
+    id2name_map = {}
+
+    # Name is a h5r object which can be searched in .mat file.
+    for index, name in enumerate(mat_file["celebrityData"]["name"][0]):
+
+        # Search real name of celebrity in .mat file.
+        id2name_map[mat_file["celebrityData"]["identity"][0][index]] = "".join(
+            chr(int(i)) for i in mat_file[name]
+        )
+    labels_map, mat_data = {}, {}
+    mat_keys_list = list(celebrity_data.keys())
```

Comment:
add variable type in variable name is not recommend, in other words, do not add a `_list` suffix to a list variable